### ISSUE TITLE
count(*) / subquery bug

### ISSUE THREAD

**Issue by nicetip** [crate/crate#7229]
**CrateDB version**: 2.3.6

**Environment description**: CentOS Linux release 7.4.1708 (Core)

**Problem description**:

`select count(*)` query returns wrong count when I use subquery internally.

Following query is OK.

```sql
cr> select count(*) from (
      select sum(cnt) from (
        select dst_ip, count(*) as cnt from logs
        where partition >= '2018-05-03' and time >= 1525311651000 and partition <= '2018-05-03' and time < 1525313451000 
        group by dst_ip
      ) 
    a) b;
+----------+
| count(*) |
+----------+
|        1 |
+----------+
```

I replaced "logs" table with subquery. Look carefully end of 3rd line.
As I know, I think it should return equal result. But it is not. 

```sql
cr> select count(*) from (
      select sum(cnt) from (
        select dst_ip, count(*) as cnt from (select * from logs) as x  -- LOOK HERE
        where partition >= '2018-05-03' and time >= 1525311651000 and partition <= '2018-05-03' and time < 1525313451000 
        group by dst_ip
      ) 
    a) b;
+----------+
| count(*) |
+----------+
|    46404 |
+----------+
```

To confirm, I queried again withoug last-wrapped query.

```sql
cr> select sum(cnt) from (
      select dst_ip, count(*) as cnt from (select * from logs) as x
      where partition >= '2018-05-03' and time >= 1525311651000 and partition <= '2018-05-03' and time < 1525313451000 
      group by dst_ip
    ) a;
+----------+
| sum(cnt) |
+----------+
| 20539926 |
+----------+
```

It's exactly 1 record.

Thanks.

**Steps to reproduce**:

**Use case**:

**Feature description**:

**Comment by chaudum** [comment]
Please, do not post screenshots of code/sql statements! They are hard to read and not portable. Github has a nice feature called Markdown support, which allows you to define code blocks in your issues and comments. [Github-flavoured Markdown](https://guides.github.com/features/mastering-markdown/) does even support fenced code blocks with syntax highlighting, e.g.

```sql
SELECT count(*) FROM table_a
```
Please, use this!

**Comment by nicetip** [comment]
Sorry, I fixed it. :)

**Comment by mfussenegger** [comment]
Thanks for reporting. We've a fix for this which will be in the next `2.3` release.
